File: Treinamento/Python/Exercícios/Estrutura_de_repetição/Questão03.py
print('*** Validação registro ***')

nome = str(input('Digite o nome desejado: '))
while len(nome) <= 3:
    print('Nome inválido, digite novamente:')
    nome = str(input('Digite o nome desejado: '))

idade = int(input('Digite a idade desejada: '))
while idade <= 0 or idade >= 150:
    print('Idade inválida, digite novamente') 
    idade = int(input('Digite a idade desejada: '))

salario = float(input('Digite o salário desejado: '))
while salario <= 0:
    print('Salário inválido, digite  novamente')
    salario = float(input('Digite o salário desejado: '))

sexo = str(input('Digite o sexo desejado: '))
while sexo[0].lower() !=  'm' and sexo[0].lower() != 'f' :
    print('Sexo inválido, digite novamente')
    sexo = str(input('Digite o sexo desejado: '))

e_civil = str(input('Digite o estado civil desejado: '))
e_civis = ['s', 'c', 'v', 'd']
e_civis2 = ['solteiro', 'casado', 'virgem', 'divorciado']
while e_civil[0].lower() not in e_civis or e_civil.lower() not in e_civis2:
    print('Estado civil inválido, digite novamente')
    e_civil = str(input('Digite o estado civil desejado: '))

print('Nome: ' + str(nome))
print('Idade: ' + str(idade) + ' anos')
print('Salário: R$' + str(salario) + ' reais')
print('E. civil: ' + str(e_civil))

    

File: Treinamento/Java/Estrutura_de_repetição/Questão03.java
import java.util.Scanner;
public class Questão03 {
    public static void main(String[] args) {
        Scanner s;
        s = new Scanner(System.in);

        System.out.print("Digite o nome desejado: ");
        String nome = s.nextLine();

        while (nome.replaceAll(" ", "").length() <= 3) {
            System.out.println("Nome inválido, digite novamente");
            System.out.print("Digite o nome desejado: ");
            nome = s.nextLine();
        } 

        System.out.print("Digite a idade: ");
        int idade = s.nextInt();

        while (idade <= 0 || idade >= 150) {
            System.out.println("Idade inválida, digite novamente");
            System.out.print("Digite a idade: ");
            idade = s.nextInt();
        }

        System.out.print("Digite seu salário: ");
        double salario = s.nextDouble();
        
        while (salario <= 0) {
            System.out.println("Salário inválido, digite novamente");
            System.out.print("Digite seu salário: ");
            salario = s.nextDouble();
        }

        System.out.print("Digite o seu sexo: ");
        String sexo = s.next();

        while (sexo.length() > 1 || (sexo.equalsIgnoreCase("m")  == false && sexo.equalsIgnoreCase("f") == false)) {
            System.out.println("Sexo, inválido, digite novamente");
            System.out.print("Digite o seu sexo: ");
            sexo = s.next();
        }
    }
}


        



        
            
            
